website_require_login/models/ir_http.py

```diff
@@ -12,13 +12,20 @@ class IrHttp(models.AbstractModel):
     @classmethod
     def _dispatch(cls):
         res = super(IrHttp, cls)._dispatch()
-
         # if not website request - skip
-
         website = request.env["website"].sudo().get_current_website()
         if not website:
             return res
-        if request.uid == website.user_id.id:
+
+        # if it can't access the user_id,
+        # it means that an exception has been
+        # raised and the cursor is currently closed
+        try:
+            user_id = website.user_id
```

Comment:
Incorrect variable name:

```suggestion
            user = website.user_id
```